Adjusting accounts 2fa (#11919)
meteor/meteor: docs/source/packages/accounts-2fa.md

Now that you have a way to allow your users to enable 2FA on their accounts, you can create a login flow based on that.

As said at the beginning of this guide, this package is currently working with two other packages: `accounts-password` and `accounts-passwordless`. Below there is an explanation on how to use this package with them.

<h3 id="working-with-accounts-password">Working with accounts-password</h3>

> When calling the function `Meteor.loginWithPassword`, if the 2FA is enabled for the user, an error will be return to the callback, so you can redirect the user to a place where they can provide a code.

Reviewer: ```suggestion
When calling the function `Meteor.loginWithPassword`, if the 2FA is enabled for the user, an error will be returned to the callback, so you can redirect the user to a place where they can provide a code.
```

As an example:

```js
<button 
  onClick={() => {
    Meteor.loginWithPassword(username, password, error => {
      if (error) {

>         if (error.reason === '2FA code must be informed') {

Reviewer: It would be better to check a code or an enum here instead of the string

Reviewer 2: I agree.

Author: :+1:

<h3 id="working-with-accounts-passwordless">Working with accounts-passwordless</h3>

Following the same logic from the previous package, if the 2FA is enabled, an error will be returned to the callback of the function [`Meteor.passwordlessLoginWithToken`](https://docs.meteor.com/packages/accounts-passwordless.html#Meteor-passwordlessLoginWithToken), then you can redirect the user to a place where they can provide a code.

> Here it's an example:

Reviewer: ```suggestion
Here is an example:
```

```js
<button
  onClick={() => {
    // logging in just with token 
    Meteor.passwordlessLoginWithToken(
      email,
      token,
      error => {
        if (error) {

>           if (error.reason === '2FA code must be informed') {

Reviewer: same here

Author: :+1:

            // send user to a page or show a component 
            // where they can provide a 2FA code
            setShouldAskCode(true);
            return;
          }

>           console.error('...', error);

Reviewer 2: Should this `console.log` be like this?

Author: It's an example... You can handle the errors any way you prefer.

2 - For the server, inside the function that will log the user in, you verify if the function `Accounts._is2faEnabledForUser` exists, and if yes, you call it providing the user you want to check if the 2FA is enabled, and if either of these statements are false, you proceed with the login flow. This function exists only when the package `accounts-2fa` is added to the project.

> If both statements are true, and the login validation was a success, now you verify if a code was provided, if not throw an error, if it was provided, verify if the code is valid by calling the function `Accounts._isTokenValid`, if not, throw an error.

Reviewer: ```suggestion
If both statements are true, and the login validation succeeds, you verify if a code was provided: if not, throw an error; if it was provided, verify if the code is valid by calling the function `Accounts._isTokenValid`. if `Accounts._isTokenValid` returns false, throw an error.
```
